src/rng/backends/mklcpu/mcg59.cpp

```diff
@@ -0,0 +1,579 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions
+* and limitations under the License.
+*
+*
+* SPDX-License-Identifier: Apache-2.0
+*******************************************************************************/
+
+#include <iostream>
+#include <CL/sycl.hpp>
+
+#include "mkl_vsl.h"
+
+#include "oneapi/mkl/exceptions.hpp"
+#include "oneapi/mkl/rng/detail/engine_impl.hpp"
+#include "oneapi/mkl/rng/detail/mklcpu/onemkl_rng_mklcpu.hpp"
+
+#include "cpu_common.hpp"
+
+namespace oneapi {
+namespace mkl {
+namespace rng {
+namespace mklcpu {

```

Comment:
I believe we can have nested namespace definition if the minimal C++ version is 17: `namespace oneapi::mkl::rng::mklcpu { ... }` if it doesn't violate any internal coding style

Reply:
done